include/CLI/App.hpp

```diff
@@ -1606,7 +1606,7 @@ class App {
 
     /// Access the config formatter as a configBase pointer
     std::shared_ptr<ConfigBase> get_config_formatter_base() const {
-        return std::dynamic_pointer_cast<ConfigBase>(config_formatter_);
+        return std::static_pointer_cast<ConfigBase>(config_formatter_);
```

Comment:
@henryiii I am not sure about this change --- is it safe?

Reply:
This is a cast up, so yes it is. Only casts down need the dynamic casts.

Reply:
I don't think this entirely safe.  ConfigBase inherits from Config.  So if someone put in a custom Config object this would be expected to return null.    On the other hand if you were doing that you should never use this function.  So it isn't unreasonable to have this function commented out in the case of turning off RTTI.

Reply:
You are correct; I chose very confusing names there and managed to confuse myself with them. Then this needs an `#if` for RTTI.

Reply:
Added a note to the doc. It you have RTTI off and have a non-standard formatter, this will return garbage.

Reply:
I think if it is a static cast it will return garbage with a custom formatter that is not based on ConfigBase regardless of RTTI.

Reply:
Might be worth renaming in CLI11 2.0 to make it a little clearer about the inheritance tree here.

Reply:
That's why I have an #if to switch it to `dynamic_cast` if RTTI is on. In 2.0, we can rename, and also just return the unmodified base pointer, and let the user do the cast.